rust/src/mqtt/parser.rs

```diff
@@ -606,7 +635,7 @@ pub fn parse_message(
                     };
                     return Ok((&input[skiplen + len..], msg));
                 }
-                MQTTTypeCode::DISCONNECT => match parse_disconnect(rem, len, protocol_version) {
+                MQTTTypeCode::DISCONNECT => match complete(parse_disconnect(len, protocol_version))(rem) {
```

Comment:
`parse_auth` is missing its `complete` addition, right ?

Reply:
Yep, done in next commit. Thanks for pointing it out.